library/ssl_srv.c

```diff
@@ -776,6 +818,24 @@ static int ssl_ciphersuite_match( mbedtls_ssl_context *ssl, int suite_id,
     }
 #endif
 
+#if defined(MBEDTLS_SSL_PROTO_TLS1_2) && \
+    defined(MBEDTLS_KEY_EXCHANGE__WITH_CERT__ENABLED)
+    /* If the ciphersuite requires signing, check whether
+     * a suitable hash algorithm is present. */
+    if( ssl->minor_ver == MBEDTLS_SSL_MINOR_VERSION_3                     &&
+        ( sig_type = mbedtls_ssl_get_ciphersuite_sig_alg( suite_info ) )
```

Comment:
Aren't we doing this regularly, e.g. with return value checks? I deliberately choose this variant here to avoid nesting conditionals.